### ISSUE TITLE
Disable user input on source selection screen until all checks for new/updated sources have completed

### ISSUE THREAD

**Issue by cckozie** [unfoldingWord-dev/ts-android#1979]
Sometimes it takes several seconds before all of the circular arrows appear. Then when I click on a checkbox with no arrow I get the dialog asking about downloading with the arrow now covered up. Other times if I click the box before the arrow appears, it moves the selected source to the Selected group without the arrow ever appearing. In that case when I click the checkbox to unselect it, it again gives me the download dialog.

**Comment by PhotoNomad0** [comment]
@ChrisJarka @cckozie - OK, but the tradeoff is that the first time you choose sources for a book, it may take a long time for the dialog to open.  Each language resource will have to be unzipped to check its date to see if it is the latest.  And in testing I found a book like Luke may have a lot of languages to check.  I found that checking about 19 languages the first time takes about 6-1/2 on a Lenovo (subsequent checks take about 20 seconds).  That can be a very frustrating user experience.  We can put up a dialog to show each language being checked, but that is still a long wait for the user.

@neutrinog

**Comment by PhotoNomad0** [comment]
@cckozie @ChrisJarka @neutrinog - maybe we could just have a spinner on each language card to indicate that we are still checking the source, but it would allow the user to still do language selection.  And they would not have to concern themselves with the languages they don't know, but focus on the language they want.

**Comment by cckozie** [comment]
@PhotoNomad0 @ChrisJarka Or possibly not check any languages (other than new ones at the bottom) until they have been selected by the user. That way they wouldn't be confused because other languages had the update arrow but theirs didn't when they clicked it. We really wouldn't lose much since there isn't any text on the screen indicating what the update arrow means, but it would be different than desktop.

**Comment by ChrisJarka** [comment]
I like the idea of the updating circle not appearing until the language is selected (from the 'Available' group to the "Selected" group).  

I just selected a source language than needed updating - said no to update. And the source language is now listed in the 'Selected' group with the circular icon. Now if I say no to the update - the language goes to the 'Available' listing.

SO - can the process from 'confirm' to the read section be stopped with a message if there needs to be downloading/updating of a language? I just confirmed 5 source languages with one with the circular icon and when I tapped the 'confirm' the window closed and the source translations that I chose were all available.  That reminder should also include a no/confirm option for those that want to still work with an old version of the source language.

The field has said that there are times that they would prefer to use an older version of the source language text.
